### PR TITLE
chore: support ts parsing for dependency test

### PR DESCRIPTION
**Pull request by asyncLiz**
Adds support for using TS syntax in https://github.com/material-components/material-components-web/blob/master/packages/material-components-web/index.ts

### CODE REVIEW

File: scripts/check-pkg-for-release.js [hunk 1]
```diff
@@ -256,11 +254,15 @@ function checkAutoInitAddedInMDCPackage(ast) {
       const callee = node.expression.callee;
       const args = node.expression.arguments;
       if (callee.object.name === 'autoInit' && callee.property.name === 'register') {
-        const expression = args.find((value) => {
-          return value.type === 'MemberExpression';
-        });
-        if (expression.object.name === nameCamel) {
-          autoInitedCount++;
+        for (let value of args) {
+          while (value.type === 'TSAsExpression') {
```
Comment (patrickrodee) [inline scripts/check-pkg-for-release.js:257]: Can you add a comment explaining why we check for the string "TSAsExpression"?

File: scripts/check-pkg-for-release.js [hunk 2]
```diff
@@ -256,11 +254,15 @@ function checkAutoInitAddedInMDCPackage(ast) {
...
+            value = value.expression;
+          }
+
+          if (value.type === 'MemberExpression' && value.object.name === nameCamel) {
```
Comment (patrickrodee) [inline scripts/check-pkg-for-release.js:261]: Can you add a comment explaining why we check for the string "MemberExpression"?

**Review by patrickrodee — APPROVED** [review]
Approved with some clarifying comment requests

File: scripts/check-pkg-for-release.js [hunk 3]
```diff
@@ -210,9 +209,7 @@ function checkJSDependencyAddedInMDCPackage() {
   const nameCamel = camelCase(CLI_PACKAGE_JSON.name.replace('@material/', ''));
   const src = fs.readFileSync(path.join(process.env.PWD, MASTER_TS_RELATIVE_PATH), 'utf8');
   const ast = recast.parse(src, {
-    parser: {
-      parse: (code) => parser.parse(code, {sourceType: 'module'}),
-    },
+    parser: require('recast/parsers/typescript'),
```
Comment (patrickrodee) [inline scripts/check-pkg-for-release.js:212]: Could we move this `require` to the top of the file and reference it below?

```
const typescriptParser = require('recast/parsers/typescript');
...
const ast = recast.parse(src, {parser: typescriptParser});
```

Comment (asyncLiz) [reply scripts/check-pkg-for-release.js]: Done

File: scripts/check-pkg-for-release.js (hunk 2 above)
Comment (asyncLiz) [reply scripts/check-pkg-for-release.js:275]: Done

File: scripts/check-pkg-for-release.js (hunk 1 above)
Comment (asyncLiz) [reply scripts/check-pkg-for-release.js:261]: Done

**Review by patrickrodee — APPROVED** [review]
🔥

**Pull request merged**